feature-libs/order/components/order-details/order-details-extended/consignment-tracking/consignment-tracking-link.component.ts

```diff
@@ -4,74 +4,57 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import {
-  ChangeDetectorRef,
-  Component,
-  ElementRef,
-  inject,
-  OnDestroy,
-  OnInit,
-  ViewChild,
-  ViewContainerRef,
-} from '@angular/core';
+import { Component, HostBinding, inject, OnInit } from '@angular/core';
 import {
   LaunchDialogService,
   LAUNCH_CALLER,
   OutletContextData,
 } from '@spartacus/storefront';
-import { of, Subscription } from 'rxjs';
-import { take } from 'rxjs/operators';
+import { Observable, of } from 'rxjs';
+import { map, take } from 'rxjs/operators';
 import { Consignment } from '@spartacus/order/root';
 import { OrderConsignmentsService } from '../../order-consignments.service';
 
 @Component({
   selector: 'cx-consignment-tracking-link',
   templateUrl: './consignment-tracking-link.component.html',
-  host: { class: 'cx-list-header col-12' },
+  // host: { class: 'cx-list-header col-12' },
 })
-export class ConsignmentTrackingLinkComponent implements OnInit, OnDestroy {
-  consignment: Consignment;
-  protected subscription = new Subscription();
+export class ConsignmentTrackingLinkComponent implements OnInit {
+  @HostBinding('className') componentClass: string;
+  consignment$: Observable<Consignment>;
   protected orderConsignmentsService = inject(OrderConsignmentsService);
   protected launchDialogService = inject(LaunchDialogService);
-  protected vcr = inject(ViewContainerRef);
-  protected cd = inject(ChangeDetectorRef);
   protected outlet = inject(OutletContextData);
 
-  @ViewChild('element') element: ElementRef;
-  consignmentStatus: string[] = this.orderConsignmentsService.consignmentStatus;
   ngOnInit(): void {
-    this.subscription.add(
-      this.outlet?.context$.subscribe((context: { item?: Consignment }) => {
-        if (context.item !== undefined) {
-          this.consignment = context.item;
-        }
-        this.cd.markForCheck();
-      })
+    this.componentClass = 'cx-list-header col-12';
+    this.consignment$ = this.outlet?.context$.pipe(
+      map((context) => context.item)
     );
   }
-  openTrackingDialog() {
+  openTrackingDialog(consignment: Consignment) {
     let tracking = {};
-    if (this.consignment.consignmentTracking) {
-      tracking = this.consignment.consignmentTracking;
+    if (consignment.consignmentTracking) {
+      tracking = consignment.consignmentTracking;
     }
     const modalInstanceData = {
       tracking$: of(tracking),
-      shipDate: this.consignment.statusDate,
+      shipDate: consignment.statusDate,
     };
 
     const dialog = this.launchDialogService.openDialog(
       LAUNCH_CALLER.CONSIGNMENT_TRACKING,
-      this.element,
-      this.vcr,
```

Comment:
[QUESTION]
Why don't we pass the element and VCR anymore? Any disadvantaves of passing them?
Are there not any advantages of passing them? e.g. keeping the dialog as a child in the change-detection tree?